src/applications/personalization/profile360/components/PaymentInformationEditModal.jsx

```diff
@@ -0,0 +1,228 @@
+        ...vals,
+      },
+    });
+  }
+
+  getRoutingNumberErrorMessage(value) {
+    if (!value.match(/^\d{9}$/)) {
+      return 'Please enter a valid routing number.';
+    }
+    return null;
+  }
+
+  getAccountNumberErrorMessage(value) {
+    if (!value.match(/^\d{1,17}$/)) {
+      return 'Please enter a valid account number.';
+    }
+    return null;
+  }
+
+  render() {
+    const editedState = this.props.paymentInformationUiState.editModalFields;
+    const lastResponse = this.props.paymentInformationUiState.response;
+
+    // The modal values may still be initializing into the store.
+    if (!editedState) return null;
+
+    return (
+      <Modal
+        title="Edit direct deposit information"
+        visible={this.props.paymentInformationUiState.isEditing}
+        onClose={this.onClose}
+      >
+        <AlertBox
+          status="error"
+          isVisible={lastResponse && lastResponse.errors.length}
+        >
+          <p>
+            We’re sorry. Something went wrong on our end and we couldn’t save
+            the recent updates you made to your profile. Please try again later.
+          </p>
+        </AlertBox>
+        <p>Update your account and routing number</p>
+        <img
+          src="/img/direct-deposit-check-guide.png"
+          alt="On a personal check, find your bank's 9-digit routing number listed along the bottom-left edge, and your account number listed beside that."
+        />
+
+        <form onSubmit={this.onSubmit}>
+          <ErrorableTextInput
+            label="Routing number (9 digits)"
+            field={editedState.financialInstitutionRoutingNumber.field}
+            errorMessage={
+              editedState.financialInstitutionRoutingNumber.errorMessage
+            }
+            onValueChange={this.onRoutingNumberChanged}
+            required
+            charMax={9}
+          />
+
+          <ErrorableTextInput
+            label="Account number (No more than 17 digits)"
+            field={editedState.accountNumber.field}
+            errorMessage={editedState.accountNumber.errorMessage}
+            onValueChange={this.onAccountNumberChanged}
+            required
+            charMax={17}
+          />
+
+          <ErrorableSelect
+            label="Account type"
+            value={editedState.accountType.value}
+            onValueChange={this.onAccountTypeChanged}
+            options={editedState.accountType.options}
+            required
+          />
+
+          <button
+            type="submit"
+            className="usa-button-primary vads-u-width--auto"
+            disabled={this.props.paymentInformationUiState.isSaving}
+          >
+            Update
+          </button>
+
+          <button
+            type="button"
+            className="usa-button-secondary"
+            onClick={this.onClose}
+          >
+            Cancel
+          </button>
+        </form>
+      </Modal>
+    );
+  }
+}
+
```

Comment:
The forms system didn't feel like a good fit for a minimal form like this, but my experience with that library is pretty slim. What do you think?

Reply:
Ah I was gonna ask about that. on the new ID form page that logged-out users see before they can fill out the HCA, I used a `SchemaForm` component directly. I'm not sure if something like that would be useful here as well. https://github.com/department-of-veterans-affairs/vets-website/blob/01f348a9b64ce9039c91534e5bd7c754fb64f337/src/applications/hca/containers/IDPage.jsx#L180

Reply:
It might be nice to use SchemaForm here as well. But half the reason to do so is to avoid the work of setting up each form field directly and you already did that :-p

Reply:
Yeah, this form is tiny, I think this approach or using SchemaForm directly seems fine. The advantage with SchemaForm is that you don't have to write a bunch of focus and validation handling yourself, you can use our well-tested forms code.

For example, the `onSubmit` method in here needs to set the focus to the first field with an error when there are outstanding errors.

Reply:
I updated `onSubmit` to set focus. I will consider switching to `SchemaForm` in a follow-up ticket but I don't think it's necessary at this time.